src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseSigner.cs

```diff
@@ -90,7 +150,7 @@ private void ValidateAlgorithmHeader(ReadOnlyMemory<byte> encodedAlg, int expect
                     exMsg = SR.Format(SR.Sign1SignCoseAlgorithmDoesNotMatchSpecifiedKeyAndHashAlgorithm, alg.Value, _keyType, HashAlgorithm.Name);
                 }
 
-                throw new CryptographicException(exMsg);
+                throw new ArgumentException(exMsg);
```

Comment:
Presumably we can always know this came from "protectedHeaders":

```suggestion
                throw new ArgumentException(exMsg, "protectedHeaders");
```